client/client.go

```diff
@@ -386,3 +410,61 @@ func stripQuotes(word string) string {
 	}
 	return word
 }
+
+// Takes raw payload and does the http request
+// - Used for login request
+// - Passwords with special chars have issues when using container
+// - For encoding/decoding
+func (c *Client) MakeRestRequestRaw(method string, rpath string, payload []byte, authenticated bool) (*http.Request, error) {
+
+	pathURL, err := url.Parse(rpath)
+	if err != nil {
+		return nil, err
+	}
+
+	fURL, err := url.Parse(c.BaseURL.String())
+	if err != nil {
+		return nil, err
+	}
+
+	if c.preserveBaseUrlRef {
+		// Default is false for preserveBaseUrlRef - matching original behavior to strip out BaseURL
+		fURLStr := fURL.String() + pathURL.String()
+		fURL, err = url.Parse(fURLStr)
+		if err != nil {
+			return nil, err
+		}
+	} else {
+		// Original behavior to strip down BaseURL
+		fURL = fURL.ResolveReference(pathURL)
+	}
+
+	var req *http.Request
+	log.Printf("[DEBUG] BaseURL: %s, pathURL: %s, fURL: %s", c.BaseURL.String(), pathURL.String(), fURL.String())
+	if method == "GET" {
+		req, err = http.NewRequest(method, fURL.String(), nil)
+	} else {
+		req, err = http.NewRequest(method, fURL.String(), bytes.NewBuffer(payload))
+	}
+	if err != nil {
+		return nil, err
+	}
+
+	if c.skipLoggingPayload {
```

Comment:
https://github.com/ciscoecosystem/aci-go-client/pull/294

will this also impact us in this client?

Reply:
Most likely.